Modules/Sources/Storage/Tools/StorageType+Extensions.swift

```diff
@@ -855,6 +855,24 @@ public extension StorageType {
         return firstObject(ofType: SystemPlugin.self, matching: predicate)
     }
 
+    /// Returns a system plugin with a specified `siteID` and `fileNameWithoutExtension`.
+    ///
+    /// - Parameters:
+    ///   - siteID: The site ID to filter by.
+    ///   - fileNameWithoutExtension: The plugin file name to match without extension (e.g., "woocommerce", "woocommerce-payments").
+    ///   - active: Optional active state filter. If provided, only plugins with matching active state are returned. If nil, active state is ignored.
+    /// - Returns: The matching system plugin, or nil if not found.
+    func loadSystemPlugin(siteID: Int64, fileNameWithoutExtension: String, active: Bool? = nil) -> SystemPlugin? {
```

Comment:
Maybe we need a backlog task for replacing all the system plugin checks with `fileNameWithoutExtension` throughout all the app for consistency?